pkg/protocol/engine/notarization/slotnotarization/slotmutations.go

```diff
@@ -61,15 +61,20 @@ func (m *SlotMutations) Evict(index iotago.SlotIndex) error {
 	return nil
 }
 
-func (m *SlotMutations) Reset(index iotago.SlotIndex) {
-	m.evictionMutex.Lock()
-	defer m.evictionMutex.Unlock()
-
-	for i := m.latestCommittedIndex; i > index; i-- {
-		m.acceptedBlocksBySlot.Delete(i)
+// Reset resets the component to a clean state as if it was created at the last commitment.
+func (m *SlotMutations) Reset() {
+	slotsToReset := make([]iotago.SlotIndex, 0)
+	m.acceptedBlocksBySlot.ForEachKey(func(slot iotago.SlotIndex) bool {
+		if slot > m.latestCommittedIndex {
+			slotsToReset = append(slotsToReset, slot)
+		}
+
+		return true
+	})
+
+	for _, slot := range slotsToReset {
+		m.acceptedBlocksBySlot.Delete(slot)
```

Comment:
delete directly

Reply:
Foreach grabs a read lock.

Reply:
Clear the whole thing as it only contains uncommitted data.